Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/AddUserController.java

```diff
@@ -26,26 +28,41 @@
 	 * 
 	 * @param req
 	 *            Holds HTTP request information
+	 * @param model
+	 * 			  Holds generic information about the model
+	 * @param user
+	 * 			  Holds the User bean values retrieved from the add page
 	 * @return String to redirect user to user list page
 	 */
 	@RequestMapping(value = "auth/user/createuser")
-	public String createUser(HttpServletRequest req) {
+	public String createUser(HttpServletRequest req, ModelMap model, @ModelAttribute("user") User user) {
 
-		User user = new User(req.getParameter("username"),
-				req.getParameter("password"));
-		user.setName(req.getParameter("fullname"));
-		user.setEmail(req.getParameter("email"));
-		usersManager.addUser(user);
+		User newUser = new User(user.getUser(), user.getPw());
+		newUser.setName(user.getName());
+		newUser.setEmail(user.getEmail());
+		if(usersManager.findUser(newUser.getUser())== null)
+			usersManager.addUser(user);
+		else

```

Comment:
You probably should do this kind of validation in a validator rather than the controller.